src/Microsoft.Tye.Core/ApplicationFactory.cs

```diff
@@ -155,6 +160,11 @@ public static async Task<ApplicationBuilder> CreateAsync(OutputContext output, F
                         throw new CommandException("Unable to determine service type.");
                     }
 
+                    service.Dependencies.AddRange(parentDependencies);
+                    parentDependencies.Add(service.Name);
```

Comment:
Seeing these two things right next to each other (inside a loop over services) makes me assume that this has an order dependency.  Maybe some examples in the comments would help? What does parent dependencies represent?